aptos-move/block-executor/src/view.rs

```diff
@@ -601,28 +607,31 @@ impl<'a, T: Transaction, S: TStateView<Key = T::Key>, X: Executable> TAggregator
         mode: AggregatorReadMode,
     ) -> anyhow::Result<aptos_aggregator::types::AggregatorValue> {
         match &self.latest_view {
-            ViewState::Sync(state) => {
-                let result = match mode {
-                    AggregatorReadMode::Aggregated => {
-                        state.read_aggregator_v2_aggregated_value(*id, self.txn_idx)
-                    },
-                    AggregatorReadMode::LastCommitted => state
-                        .read_aggregator_v2_last_committed_value(
-                            *id,
-                            self.txn_idx,
-                            ReadPosition::BeforeCurrentTxn,
-                        ),
-                };
-                result.map_err(|e| {
-                    anyhow::Error::new(VMStatus::error(
-                        StatusCode::STORAGE_ERROR,
-                        Some(format!("Error during read: {:?}", e)),
-                    ))
-                })
+            ViewState::Sync(state) => match mode {
+                AggregatorReadMode::Aggregated => state
+                    .read_aggregator_v2_aggregated_value(*id, self.txn_idx)
+                    .map_err(|e| {
+                        anyhow::Error::new(VMStatus::error(
+                            StatusCode::from(&e),
+                            Some(format!("Error during read: {:?}", e)),
+                        ))
+                    }),
+                AggregatorReadMode::LastCommitted => state
+                    .read_aggregator_v2_last_committed_value(
+                        *id,
+                        self.txn_idx,
+                        ReadPosition::BeforeCurrentTxn,
+                    )
+                    .map_err(|e| {
```

Comment:
I hope we can still uniformize the treatment of these errors? (see my related comment in the versioned aggregators).